File: src/SlidingApps.TaskRunner.Web/app/service/write-model/tenant/create-tenant.ts
/// <reference path="../../../typings.d.ts" />

// MODEL
import { IPayload } from '../../../component/foundation/model';

export interface ICreateTenantPayload extends IPayload {
    code: string;
    name: string;
    description: string;
    validFrom?: string;
    validUntil?: string;
    userName: string;
    userPassword: string;
}

File: src/SlidingApps.TaskRunner.WriteModel.Platform.Domain.Model/Tenants/Intents/CreateTenant.cs

using SlidingApps.TaskRunner.Foundation.Cqrs;
using System;

namespace SlidingApps.TaskRunner.WriteModel.Platform.Domain.Model.Tenants.Intents
{
    public class CreateTenant
        : IIntent
    {
        public string Code { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public DateTime? ValidFrom { get; set; }

        public DateTime? ValidUntil { get; set; }

        public string UserName { get; set; }

        public string UserPassword { get; set; }
    }
}
